### PR TITLE
add inventory policy

### PR DESCRIPTION
**Pull request by Liujingfang1**
This change adds the option InventoryPolicy. This policy controls the behavior of apply/prune behavior. This control is done by checking if an apply/prune operation can be taken for an object with a given InventoryInfo. 

This change covers the part that if an operation can go through. In the next change, this check will be applied in the Applier.

### CODE REVIEW

File: pkg/inventory/policy.go [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+const owningInventoryKey = "config.kubernetes.io/owning-inventory"
+
+// inventoryIDMatchStatus represents the result of comparing the
+// id from current inventory info and the inventory-id from a live object.
+type inventoryIDMatchStatus int
+
+const (
+	Empty inventoryIDMatchStatus = iota
+	Match
+	Unmatch
```
Comment (mortent) [inline pkg/inventory/policy.go:73]: I think `NoMatch` might be a better name here?

Comment (Liujingfang1) [reply pkg/inventory/policy.go]: updated

File: pkg/inventory/policy.go [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+	// The apply operation can go through for any resource in the package even if the
+	// live object has an unmatched inventory-id annotation.
+	//
+	// The prune operation can go through when
+	// - The inventory-id annotation in the live object match with that
+	//   in the package.
+	AdoptAll
+)
+
+const owningInventoryKey = "config.kubernetes.io/owning-inventory"
```
Comment (mortent) [inline pkg/inventory/policy.go:64]: We should revisit the name of this annotation, ref https://github.com/GoogleContainerTools/kpt/issues/1257

Comment (Liujingfang1) [reply pkg/inventory/policy.go]: changed it to "config.k8s.io/owning-inventory"

**Comment by Liujingfang1** [comment]
Addressed the comments. It's ready for another review.

**Review by mortent — APPROVED** [review]
/lgtm

**Pull request merged**